modules/tinymce/src/core/main/ts/api/commands/ClipboardCommands.ts

```diff
@@ -1,37 +1,94 @@
+/* eslint-disable no-console */
 import Editor from '../Editor';
 import Env from '../Env';
 
 export const registerCommands = (editor: Editor): void => {
-  editor.editorCommands.addCommands({
-    'Cut,Copy,Paste': (command) => {
-      const doc = editor.getDoc();
-      let failed;
-
-      // Try executing the native command
-      try {
-        doc.execCommand(command);
-      } catch (ex) {
-        // Command failed
-        failed = true;
-      }
 
-      // Chrome reports the paste command as supported however older IE:s will return false for cut/paste
-      if (command === 'paste' && !doc.queryCommandEnabled(command)) {
-        failed = true;
-      }
+  const copyToClipboard = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text);
+      console.log('copyToClipboard', text);
+    } catch (err) {
+      handleClipboardError();
+    }
+  };
 
-      // Present alert message about clipboard access not being available
-      if (failed || !doc.queryCommandSupported(command)) {
-        let msg = editor.translate(
-          `Your browser doesn't support direct access to the clipboard. ` +
-          'Please use the Ctrl+X/C/V keyboard shortcuts instead.'
-        );
+  const pasteFromClipboard = async () => {
+    let text = '';
 
-        if (Env.os.isMacOS() || Env.os.isiOS()) {
-          msg = msg.replace(/Ctrl\+/g, '\u2318+');
-        }
+    try {
+      text = await navigator.clipboard.readText();
+      console.log('pasteFromClipboard', text);
+      editor.selection.setContent(text);
+    } catch (err) {
+      handleClipboardError();
+    }
+    return text;
+  };
 
-        editor.notificationManager.open({ text: msg, type: 'error' });
+  const cutToClipboard = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text);
+      console.log('cutToClipboard', text);
+      editor.selection.setContent('');
+    } catch (err) {
+      handleClipboardError();
+    }
+  };
+
+  const fallbackClipboardCommand = (editor: Editor, command: string) => {
+    const doc = editor.getDoc();
+    let failed = false;
+
+    try {
+      doc.execCommand(command);
```

Comment:
There is no point to this code anymore, the only browser it ever worked on was IE.